### PR TITLE
[Core][Geometry] QuadraturePointGeometry

### PR DESCRIPTION
**Pull request by tteschemacher**
Hi all,

A PR based on a PR which is based on another one. Very beautiful ;).

This is a geometry which acts as a single integration/ quadrature point. To avoid naming conflicts it is called quadrature point. This is essential to combine different discretizations but still using the same geometry. This is one of the new geometries discussed in #5042 .
It can be used in IGA with high order, but also for linear standard geometries.

Additionally, also test for the serialization are added.

Cheers!

### CODE REVIEW

**Review by loumalouomega — COMMENTED** [review]
OK from my side

File: kratos/tests/cpp_tests/geometries/test_quadrature_point.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:     BSD License
+//           Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
```
Comment (philbucher) [inline kratos/tests/cpp_tests/geometries/test_quadrature_point.cpp]: :v:

File: kratos/tests/cpp_tests/sources/test_serializer_geometry_data.cpp [hunk 2]
```diff
@@ -0,0 +1,158 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:     BSD License
+//           Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
```
Comment (philbucher) [inline kratos/tests/cpp_tests/sources/test_serializer_geometry_data.cpp]: :v:

File: kratos/geometries/quadrature_point.h [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+    //{
+    //    mpGeometryParent = pGeometryParent;
+    //}
+
+    ///@}
+    ///@name Geometrical Operations
+    ///@{
+
+    /// Returns the domain size of this quadrature point.
+    double DomainSize() const override
```
Comment (philbucher) [inline kratos/geometries/quadrature_point.h]: how can the domain size be a double? :confused:

**Comment by loumalouomega** [comment]
DomainSize is the volume in 3D and area en 2D

**Comment by adityaghantasala** [comment]
Domain size is 2 or 3 right ? an Int ?

**Comment by tteschemacher** [comment]
No, domain size is the respective space of each geometry.

@adityaghantasala I think you mean Dimension() maybe?

File: kratos/tests/cpp_tests/geometries/test_quadrature_point.cpp (hunk 1 above)
Comment (tteschemacher) [reply kratos/tests/cpp_tests/geometries/test_quadrature_point.cpp]: Man, who reads the Authors of a file? :P

**Comment by adityaghantasala** [comment]
I am confused because, in the input JSON we have this

`"domain_size": 3,`

so I thought this is that ... ! :-|

**Comment by tteschemacher** [comment]
Actually, this is really confusing. However, this domain size is something else. This one is for how many historic time steps are kept. The one from the geometry tells about the spatial size.

**Comment by tteschemacher** [comment]
This is actually the WorkingSpaceDimension().

**Comment by pooyan-dadvand** [comment]
> I am confused because, in the input JSON we have this
> 
> `"domain_size": 3,`
> 
> so I thought this is that ... ! :-|

Indeed it is really confusing

**Comment by philbucher** [comment]
this gives a coredump in Travis

**Comment by tteschemacher** [comment]
The core dump comes from the serializer. Whilest loading the shapefunctioncontainer.

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee approves the QuadraturePoint class and the new `GetParentGeometry()` and `SetParentGeometry()` to the base class throwing an error

**Comment by tteschemacher** [comment]
Can you approve @philbucher ? All changes are now as requested.

**Review by philbucher — APPROVED** [review]

**Comment by tteschemacher** [comment]
thx

**Comment by pooyan-dadvand** [comment]
@tteschemacher would you please change the method names as suggested?

Also would you please make the current Get method returning a const reference? And if you need a non const reference you may add a non const version  of Get method returning a normal reference.

@philbucher would explain you the details.

**Pull request merged**